e2e/test/provisioning/ProvisioningE2ETests.cs

```diff
@@ -1090,28 +1092,9 @@ private static void ValidateDeviceRegistrationResult(bool validatePayload, Devic
             VerboseTestLogger.WriteLine($"{result.Status} (Error Code: {result.ErrorCode}; Error Message: {result.ErrorMessage})");
             VerboseTestLogger.WriteLine($"ProvisioningDeviceClient AssignedHub: {result.AssignedHub}; DeviceID: {result.DeviceId}");
 
-            result.Status.Should().Be(ProvisioningRegistrationStatusType.Assigned, $"Unexpected provisioning status, substatus: {result.Substatus}, error code: {result.ErrorCode}, error message: {result.ErrorMessage}");
+            result.Status.Should().Be(ProvisioningRegistrationStatus.Assigned, $"Unexpected provisioning status, substatus: {result.Substatus}, error code: {result.ErrorCode}, error message: {result.ErrorMessage}");
             result.AssignedHub.Should().NotBeNull();
             result.DeviceId.Should().NotBeNull();
-
-            if (validatePayload)
```

Comment:
unreachable code? If yes, then we can probably trim the params that this method takes in.